x/superfluid/keeper/migrate_test.go

```diff
@@ -1230,179 +1231,195 @@ type positionInfo struct {
+				coin1Amt := DefaultAmt1.QuoRaw(divisor)     // Divide amount1 by the divisor to add some entropy to the position creation
+				positionCoins := sdk.NewCoins(sdk.NewCoin(DefaultCoin0.Denom, coin0Amt), sdk.NewCoin(DefaultCoin1.Denom, coin1Amt))
+				s.FundAcc(s.TestAccs[index], positionCoins)
+				totalFundsForPositionCreation = totalFundsForPositionCreation.Add(positionCoins...) // Track total funds used for position creation, to be used by invariant checks later
+				posInfoInternal := s.createBalancerPosition(s.TestAccs[index], balancerPoolId, lockDurationFn(i), balancerPoolShareDenom, positionCoins, positions.numAccounts-i, superfluidDelegate)
+				positionInfos[positionType] = append(positionInfos[positionType], posInfoInternal) // Track position info for invariant checks later
+				followOnFn(index, posInfoInternal)
+			}
+		}
 
-	for i := 0; i < positions.numUnbondingSuperfluidUnlocking; i++ {
-		posInfoInternal := s.createBalancerPosition(s.TestAccs[i+1], balancerPoolId, unbondingDuration+time.Nanosecond*2, balancerPoolShareDenom, DefaultCoins, positions.numAccounts-i, true)
-		err := s.App.SuperfluidKeeper.SuperfluidUndelegate(s.Ctx, s.TestAccs[i+1].String(), posInfoInternal.lockId)
-		s.Require().NoError(err)
-		err = s.App.SuperfluidKeeper.SuperfluidUnbondLock(s.Ctx, posInfoInternal.lockId, s.TestAccs[i+1].String())
-		s.Require().NoError(err)
-		positionInfos[2] = append(positionInfos[2], posInfoInternal)
-	}
+		createPositions(0, positions.numBondedSuperfluid, func(int) time.Duration { return unbondingDuration }, true, func(index int, posInfoInternal positionInfo) {
+		})
 
-	for i := 0; i < positions.numVanillaLockLocked; i++ {
-		posInfoInternal := s.createBalancerPosition(s.TestAccs[i+1], balancerPoolId, unbondingDuration+time.Nanosecond*3, balancerPoolShareDenom, DefaultCoins, positions.numAccounts-i, false)
-		positionInfos[3] = append(positionInfos[3], posInfoInternal)
-	}
+		createPositions(1, positions.numUnbondingSuperfluidLocked, func(int) time.Duration { return unbondingDuration + time.Nanosecond }, true, func(index int, posInfoInternal positionInfo) {
+			err := s.App.SuperfluidKeeper.SuperfluidUndelegate(s.Ctx, s.TestAccs[index].String(), posInfoInternal.lockId)
+			s.Require().NoError(err)
+		})
 
-	for i := 0; i < positions.numVanillaLockUnlocking; i++ {
-		posInfoInternal := s.createBalancerPosition(s.TestAccs[i+1], balancerPoolId, unbondingDuration+time.Nanosecond*4, balancerPoolShareDenom, DefaultCoins, positions.numAccounts-i, false)
-		lock, err := s.App.LockupKeeper.GetLockByID(s.Ctx, posInfoInternal.lockId)
-		s.Require().NoError(err)
-		_, err = s.App.LockupKeeper.BeginUnlock(s.Ctx, posInfoInternal.lockId, lock.Coins)
-		s.Require().NoError(err)
-		positionInfos[4] = append(positionInfos[4], posInfoInternal)
-	}
+		createPositions(2, positions.numUnbondingSuperfluidUnlocking, func(int) time.Duration { return unbondingDuration + time.Nanosecond*2 }, true, func(index int, posInfoInternal positionInfo) {
```

Comment:
nit: maybe use enums for code sanity for the first argument

Reply:
enums added here https://github.com/osmosis-labs/osmosis/pull/5565/commits/62fdfa5a88fa4dbeb6baa58a15d09345fdac5716